prescient/engine/egret/engine.py

```diff
@@ -233,33 +260,64 @@ def create_and_solve_lmp(self,
-                                                     ptdf_options = self._ptdf_manager.lmpsced_ptdf_options,
-                                                     PTDF_matrix_dict=self._ptdf_manager.PTDF_matrix_dict)
-
-        self._p._zero_out_costs(pyo_model, self._hours_in_objective)
+        if self._last_sced_pyo_model is None:
+            self._ptdf_manager.mark_active(lmp_sced_instance)
+            pyo_model = create_sced_uc_model(lmp_sced_instance, relaxed=True,
+                                             ptdf_options = self._ptdf_manager.lmpsced_ptdf_options,
+                                             PTDF_matrix_dict=self._ptdf_manager.PTDF_matrix_dict)
+            pyo_solver = self._sced_solver
+            self._p._zero_out_costs(pyo_model, self._hours_in_objective)
+        else:
+            pyo_model = self._last_sced_pyo_model
+            pyo_solver = self._last_sced_pyo_solver
+            self._transform_for_lmp(pyo_model, pyo_solver, lmp_sced_instance)
 
         pyo_model.dual = pe.Suffix(direction=pe.Suffix.IMPORT)
 
         try:
-            lmp_sced_results, _ = self._p.call_solver(self._sced_solver, 
-                                                   pyo_model,
-                                                   options,
-                                                   options.sced_solver_options,
-                                                   relaxed=True)
+            lmp_sced_results, _, _ = self._p.call_solver(pyo_solver,
+                                                         pyo_model,
+                                                         options,
+                                                         options.sced_solver_options,
+                                                         relaxed=True,
+                                                         set_instance=(self._last_sced_pyo_model is None))
         except:
             print("Some issue with LMP SCED, writing instance")
             quickstart_uc_filename = options.output_directory+os.sep+"FAILED_LMP_SCED.json"
             lmp_sced_instance.write(quickstart_uc_filename)
             print(f"Problematic LMP SCED written to {quickstart_uc_filename}")
             raise
 
-        self._ptdf_manager.update_active(lmp_sced_results)
         self._attach_fake_pyomo_objects(lmp_sced_results)
 
         return lmp_sced_results
 
+    def _transform_for_lmp(self, pyo_model, pyo_solver, lmp_sced_instance):
+        from pyomo.solvers.plugins.solvers.persistent_solver import PersistentSolver
+        import math
+
+        uc_instance_binary_relaxer(pyo_model, pyo_solver)
+
+        ## reset the penalites
+        system = lmp_sced_instance.data['system']
+
+        update_obj = False
+
+        new_load_penalty = system['baseMVA'] * system['load_mismatch_cost']
+        if not math.isclose(new_load_penalty, pyo_model.LoadMismatchPenalty.value):
```

Comment:
Nit - I think we're trying to standardize people on "pyo.value(foo)" instead of "foo.value".

Reply:
We explicitly rely on the fact that this Param is mutable on the following line by setting the value attribute, though I'm not necessarily opposed to using pyo.value here either.

Reply:
Agree - I just wanted to point out the "NPS" standard for reading params and vars, whether the former are mutable or not.